SerialPrograms/Source/Integrations/DppIntegration/DppCommandHandler.cpp

```diff
@@ -0,0 +1,399 @@
+                            cmds.emplace_back(element.second.command);
+                        }
+                        bot.global_bulk_command_create(cmds);
+                        m_initialized = true;
+                    }
+                });
+
+                bot.on_guild_create([&bot, this](const guild_create_t& event) {
+                    try {
+                        std::string id = std::to_string(event.created->id);
+                        this->log_dpp("Loaded guild: " + event.created->name + " (" + id + ").", "Guild Create", ll_info);
+                        std::lock_guard<std::mutex> lg(m_count_lock);
+                        Utility utility;
+                        std::thread(&Utility::get_user_counts, std::ref(utility), std::ref(bot), event).detach();
+                    }
+                    catch (std::exception& e) {
+                        this->log_dpp("Failed to get user counts: " + (std::string)e.what(), "Guild Create", ll_error);
+                    }
+                });
+
+                bot.on_guild_member_add([this](const guild_member_add_t& event) {
+                    std::string id = std::to_string(event.adding_guild->id);
+                    if (!user_counts.empty() && user_counts.count(id)) {
+                        this->log_dpp("New member joined " + event.adding_guild->name + ". Incrementing member count.", "Guild Member Add", ll_info);
+                        user_counts.at(id)++;
+                    }
+                });
+
+                bot.on_guild_member_remove([this](const guild_member_remove_t& event) {
+                    std::string id = std::to_string(event.removing_guild->id);
+                    if (!user_counts.empty() && user_counts.count(id)) {
+                        this->log_dpp("Member left " + event.removing_guild->name + ". Decrementing member count.", "Guild Member Remove", ll_info);
+                        user_counts.at(id)--;
+                    }
+                });
+
+                bot.on_slashcommand([this](const slashcommand_t& event) {
+                    auto it = command_map.find(event.command.get_command_name());
+                    if (it != command_map.end()) {
+                        this->log_dpp("Executing slash command: " + event.command.get_command_name() + ".", "Slash Command", ll_info);
+                        try {
+                            it->second.func(event);
+                        }
+                        catch (std::exception& e) {
+                            this->log_dpp("Exception occurred while executing command: " + event.command.get_command_name() + ".\n" + e.what(), "Slash Command", ll_critical);
+                        }
+                    }
+                });
+            }
+
+            void Handler::send_message(cluster& bot, embed& embed, const std::string& channel, std::chrono::milliseconds delay, const std::string& msg, std::shared_ptr<PendingFileSend> file) {
```

Comment:
No, I passed the argument but haven't implemented the `ScheduledTaskRunner`. Mainly because I forgot, but DPP also sends asynchronously, so not sure if as necessary as with Sleepy. Though can't hurt.